python-api/src/main/python/api/client_builder.py

```diff
@@ -0,0 +1,77 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from ConfigParser import SafeConfigParser
+import inspect
+from StringIO import StringIO
+from urlparse import urlparse
+import client_factory
+
+class ClientBuilder(object):
+
+    _CONFIG_SECTION = 'env'
+
+    def __init__(self, load_defaults=True):
+        self._config = SafeConfigParser()
+        if load_defaults is True:
+            self._handle_default_config_loading()
+        else:
+            self._config.add_section(self._CONFIG_SECTION)
+
+    def _handle_default_config_loading(self):
+        config_files = ['spark-defaults.conf', 'livy-client.conf']
+        for config_file in config_files:
+            with open(config_file) as stream:
```

Comment:
You should re-think about how to do configuration in the python API. You're passing a relative path to `open`, which means you'll be loading things from the current directory. That is definitely not what I'd expect.

I don't believe python has an equivalent of `Class.getResource()` in Java, so maybe the API will have to be more explicit about where to find config data, or look at environment variables, or something else.